libs/fileset/glob.go

```diff
@@ -0,0 +1,45 @@
+package fileset
+
+import (
+	"io/fs"
+	"os"
+	"path/filepath"
+)
+
+type GlobSet struct {
+	root     string
+	patterns []string
+}
+
+func NewGlobSet(root string, includes []string) *GlobSet {
+	for k := range includes {
+		includes[k] = filepath.Join(root, filepath.FromSlash(includes[k]))
+	}
+	return &GlobSet{root, includes}
```

Comment:
In the set key function we depend on the root being an absolute path. Can you double check that here, so that the file matches we return from `All()` have in fact an absolute path?

If you pass e.g. `../foo` as root then they won't be.

Reply:
@pietern thanks for the catch, fixed by ensuring root is always absolute and added unit test